fix(dashboard): update  cross filter scoping chart id references during dashboard import

chanduapple opened the pull request:
during the dashboard import, cross filter scoping is not retained because of not updating chart id references. this commit contains the necessary fix for the same. 

this  pull request provides the fix for the issue 
[#19944 ](https://github.com/apache/superset/issues/19944)
slack discussion: https://apache-superset.slack.com/archives/C015WAZL0KH/p1699248398108929

### SUMMARY
updated the utils.py to update the chart id references for cross filter scoping also. currently chart id reference updates is present for native filters, position and few others, but not for cross filter scoping.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
1. create dashboard with more than 2 charts and apply cross filter scoping.  as shown below. 
2. export the dashboard.
3. if you are importing in the same environment, then delete dashboard and charts. 
4. import the dashboard again. 
5. expectation is cross filter scoping should be retained. 
6. actual is cross filter scoping has been removed and global cross filter scoping changes to all charts. 
dashboard in env1 
![image](https://github.com/apache/superset/assets/80615671/2ff9b3b6-b3ef-4084-859a-dad530568c28)

# BEFORE FIX 
dashboard after importing to env2
![image](https://github.com/apache/superset/assets/80615671/4a039524-730e-4490-aa2f-817fdd3db924)

# AFTER FIX
dashboard after importing to env2
![image](https://github.com/apache/superset/assets/80615671/d5d01b01-a4ad-413a-944c-e09bc97a9240)



### TESTING INSTRUCTIONS
1. create dashboard with cross filter scoping.
2. export the dashboard.
3. delete the dashboard and charts used in dashboards.
4. import the dashboard
5. check the cross filter scoping

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
Fixes "#19944" 
- [x] Has associated issue: #19944 
- [ ] Required feature flags: 
- [ ] Changes UI: 
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

chanduapple committed:
updated chart id references for cross filter scoping

during the dashboard import, cross filter scoping is not retained because of not updating chart id references. this commit contains the necessary fix for the same

chanduapple commented:
@john-bodley @michael-s-molina @villebro  could you please review this PR. This is kind of blocker to use import/export feature for dashboard, since people need to make the changes manually after importing dashboard as cross filter scoping was not retained. 
i am contributing first time, please let me know if any changes required or to furnish more info.

villebro commented:
@betodealmeida do you think it would make sense to start phasing both the pseudo-ids that are used on the frontend, and the serial ids used on the backend, and replace them all with their respective uuid representations? This way we wouldn't have to do all this complicated remapping during import/export..

villebro commented:
> @john-bodley @michael-s-molina @villebro could you please review this PR. This is kind of blocker to use import/export feature for dashboard, since people need to make the changes manually after importing dashboard as cross filter scoping was not retained. i am contributing first time, please let me know if any changes required or to furnish more info.

@chanduapple thanks for the contribution, and I agree with the changes done here. I'll let CI complete before reviewing the code. Also, I'm not sure if our current test suites have tests for similar re-mappings (the native filters may have one), but if it's possible, it would be great if you could add a test to cover this particular fix.

chanduapple commented:
@villebro is there any guideline/info about adding tests, like how to add test and run, I will go through once, I have tested manually.

villebro commented:
@chanduapple the `CONTRIBUTING.md` has some directives for running the test suites: https://github.com/apache/superset/blob/master/CONTRIBUTING.md#testing The relevant tests I think you're interested in are in https://github.com/apache/superset/blob/master/tests/integration_tests/import_export_tests.py.

chanduapple committed:
Merge branch 'apache:master' into master

chanduapple committed:
fixed the List Error

chandrasekharjandhyam committed:
fixed Too many branches (18/15) issue

added a new function to separate the cross filter scoping fixes

chandrasekharjandhyam committed:
Revert "fixed Too many branches (18/15) issue"

This reverts commit efd13a09223212094f8d7a0546cc245b6d75126c.

chandrasekharjandhyam committed:
Revert "fixed the List Error"

This reverts commit 881219f0526ec42e85766aa70e2cda0e4dd67b29.

chandrasekharjandhyam committed:
fixed CI workflow issues

1. List error
2. fixed Too many branches (18/15)
3. fixed trailing white spaces

chandrasekharjandhyam committed:
removed the old code

chanduapple commented:
@villebro i have fixed below earlier CI workflow issues. 
1. pull request title has been modified.
2. removed trailing white spaces.
3. List error
4. R0912: Too many branches (18/15) (too-many-branches) 

Could you please approve the workflow to be rerun and review the code changes?

@betodealmeida @john-bodley

chanduapple committed:
Merge branch 'apache:master' into master

chanduapple commented:
@villebro @betodealmeida  could you please review and approve, let me know if any thing to be modified. please help me out here.

rusackas commented:
Looks like this one needs the [pre-commit hooks ](https://superset.apache.org/docs/contributing/hooks-and-linting/) run. Hope we can get it through and close out the old issue!

ferschubert commented:
Any updates on this folks? We are blocked by this issue to export->import charts with cross-filter scoping.

rusackas commented:
Not sure if anyone has interest in rebasing this and getting it in a mergeable state, but we're open to that, or others adopting the work on a fresh PR. Meanwhile I'll convert this to a draft since it's been silent for so long, and it may eventually be closed due to inactivity.

rusackas commented:
Closing due to inactivity. Feel free to reopen if/when warranted.
